cmd/vochaintest/vochaintest.go

```diff
@@ -237,6 +263,108 @@ func censusImport(host string, signer *ethereum.SignKeys) {
 	log.Infof("Census created and published\nRoot: %s\nURI: %s", root, uri)
 }
 
+func waitUntilNextBlock(mainClient *client.Client) error {
+	h, err := mainClient.GetCurrentBlock()
+	if err != nil {
+		return fmt.Errorf("cannot get current height: %w", err)
+	}
+	mainClient.WaitUntilBlock(h + 1)
+	return nil
+}
+
+func initAccounts(treasurer, oracle, mainSigner *ethereum.SignKeys, host string) error {
+	var err error
+	log.Infof("connecting to main gateway %s", host)
+	// connecting to endpoint
+	var mainClient *client.Client
+	for tries := 10; tries > 0; tries-- {
+		mainClient, err = client.New(host)
+		if err == nil {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer mainClient.Close()
+	// get the chain ID for signing the transactions
+	chainID, err := mainClient.GetChainID()
+	if err != nil {
+		log.Fatal(err)
+	}
+	// set chain id
+	treasurer.VocdoniChainID = chainID
+	oracle.VocdoniChainID = chainID
+	mainSigner.VocdoniChainID = chainID
+
+	retries := 5
+	// create and top up accounts
+	if err := ensureAccountExists(mainClient, oracle, retries); err != nil {
+		log.Fatal(err)
+	}
+	if err := ensureAccountExists(mainClient, mainSigner, retries); err != nil {
+		log.Fatal(err)
+	}
+
+	if err := ensureAccountHasTokens(mainClient, treasurer, oracle, 10000, retries); err != nil {
+		log.Fatalf("cannot mint tokens for account %s with treasurer %s: %w", oracle.Address(), treasurer.Address(), err)
+	}
+	if err := ensureAccountHasTokens(mainClient, treasurer, mainSigner, 10000, retries); err != nil {
+		log.Fatalf("cannot mint tokens for account %s with treasurer %s: %w", mainSigner.Address(), treasurer.Address(), err)
+	}
+
+	return nil
+}
+
+func ensureAccountExists(mainClient *client.Client, account *ethereum.SignKeys, retries int) error {
+	for i := 0; i <= retries; i++ {
+		// if account exists, we're done
+		if acct, _ := mainClient.GetAccount(nil, account.Address()); acct != nil {
+			return nil
+		}
+		if i == retries { // that was the last chance, break and fail immediately
+			break
+		}
+
+		if err := mainClient.CreateOrSetAccount(account, common.Address{}, "ipfs://", 0, nil); err != nil {
+			if strings.Contains(err.Error(), "tx already exists in cache") {
+				// don't worry then, someone else created it in a race, nevermind, job done.
+			} else {
+				return fmt.Errorf("cannot create account %s: %s", account.Address(), err)
+			}
+		}
+
+		waitUntilNextBlock(mainClient)
+	}
+	return fmt.Errorf("cannot create account %s after %d retries", account.Address(), retries)
+}
+
+func ensureAccountHasTokens(mainClient *client.Client, treasurer, account *ethereum.SignKeys, amount uint64, retries int) error {
```

Comment:
So yeah, following the comment I did on the unresolved thread ... You can see in this function that we are passing account as *ethereum.SignKeys but only account.Address() is used. So instead of passing the whole account signer is enough to pass the address of the account.

From `func ensureAccountHasTokens(mainClient *client.Client, treasurer, account *ethereum.SignKeys, amount uint64, retries int) error` to `func ensureAccountHasTokens(mainClient *client.Client, treasurer *ethereum.SignKeys, account common.Address, amount uint64, retries int) error`

Reply:
right, so true. pushed this change just now, thanks